Adjust to file upload database schema (not include upload form UI)

samuraikun opened the pull request:
# Description
* Upload image in Trip page and Activity page.
* Use Supabase storage

## **Important!!**

- I plan to develop upload form UI after this PR 🙏  
  - Because, this PR is too long for status and has many changes. 🙇 

## Related PR
https://github.com/tabi-memo/backend/pull/15

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Upload flow

```mermaid
sequenceDiagram
  Client ->> Supabase Storage: Direct upload
  Supabase Storage ->> Client: Return file path
  Client ->> Server Action: Update file metadata to DB
  Server Action ->> DB: Insert public URL to Trip & Activity image_url column
  Client ->> Client: Display uploaded image
```

## Check List
- Trip
  - [x] Upload image file using draft input field
  - [x] Replace another image to existing uploaded image
- Activity
  - [ ] Upload image file using exisiting input field
  - [ ] Replace another image to existing uploaded image

## URL

## Screenshot

### Supabase Storage Console

![Screenshot 2024-01-20 at 19 30 57](https://github.com/tabi-memo/frontend/assets/7115171/fdfa01ab-18a4-4d5f-9314-3601f9431ee0)

samuraikun committed:
wip: add func to upload file.

samuraikun committed:
Refactor trip image upload and display

samuraikun committed:
Merge branch 'main' into file-upload

samuraikun committed:
Add image refresh on trip detail page and trip list page after upload image.

samuraikun committed:
fix: Refactoring upload impl using Server action.

samuraikun committed:
Rename server action

YoshinariYamanaka commented:
Overall looks nice ! Especially the sequence diagram looks nice.

@samuraikun @Kana-TAGUCHI @SachiGoto 
And some questions ! 
Do we need additional meta info shown below or not ?

- Don't we need to regulate acceptable file type ?
example) We'll accept only `JPEG`, `PNG`, `SVG`. I'd say `images/*` is so wide, isn't it ...? I'm not familiar with UI, so if we can manage, fine !

- Max file size (bytes size) 
- File resize (pixels) 
Do we need to resize files or display them as is ?

kanatagu reviewed (approved):
Thank you for implementing for frontend side too! Looks good overall :)

In this case, if the user want to change the image, the old image remains in the storage right? Is that fine? Or we should upload image when the user actually post/put the trip/activity data?

samuraikun committed:
wip: add Mutation to create Activity

samuraikun committed:
Merge branch 'main' into file-upload

samuraikun committed:
Replace image_storage_object_id to image_url.
memo: Due to working app, this commit adjusts to the backend pull request of "https://github.com/tabi-memo/backend/pull/15"

samuraikun committed:
Fix for eslint

samuraikun committed:
Update image storage object ID to image URL in trip details

kanatagu reviewed (approved):
Thank you!
